tests/migrations/test_state.py

```diff
@@ -870,6 +871,37 @@ class Meta:
         self.assertEqual([name for name, mgr in food_state.managers], ['food_mgr'])
         self.assertEqual(food_state.managers[0][1].args, ('a', 'b', 1, 2))
 
+    def test_metaclass_conflict(self):
+        """
+        Test making class with complex metadata inheritance
+        """
+        new_apps = Apps(['migrations'])
+
+        class DummyMetaClass(type):
+            pass
+
+        class Mixin(six.with_metaclass(DummyMetaClass, object)):
+            mixin_attr = True
+
+        class ItermidiateMetaClass(ModelBase, DummyMetaClass):
+            pass
+
+        # Create the same Model without metaclassmaker as you would normally do.
+        class MetaclassTestModel(six.with_metaclass(ItermidiateMetaClass, models.Model, Mixin)):
```

Comment:
Doesn't matter with this `Mixin`, as it's just example of inheritance, name it and order as you like. I really wasn't thinking that much writing it ;). I know Python is calling super from left to right, but it depends on your specific case what order you would like to have. In this example it really doesn't matter.
If you insist I can change the order.